app/commands/solution/publish.rb

```diff
@@ -10,7 +10,10 @@ def call
       return if solution.published?
 
       ActiveRecord::Base.transaction do
-        solution.update(published_at: Time.current)
+        solution.update(
+          published_at: Time.current,
+          allow_comments: user_preferences.allow_comments_by_default
```

Comment:
```suggestion
          allow_comments: user.preferences.allow_comments_by_default
```

Reply:
@dem4ron Change `delegate :exercise, to: :solution` to `delegate :exercise, :user, to: :solution` below, and change any `solution.user` to `user` etc